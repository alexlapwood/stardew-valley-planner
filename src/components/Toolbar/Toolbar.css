.Toolbar {
  bottom: 32px;
  left: 50%;
  position: absolute;
  transform-origin: 0 0;
  transform: scale(1.5) translateX(-50%);
}

.Toolbar--tool {
  height: 16px;
  width: 16px;
}
