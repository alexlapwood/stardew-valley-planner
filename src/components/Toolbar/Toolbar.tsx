import * as React from "react";

import * as cn from "classnames";

import "./Toolbar.css";

interface IProps {
  images: HTMLImageElement[];
  selectedToolId?: string;
  selectTool: (tool: string) => void;
}

const Toolbar: React.SFC<IProps> = props => (
  <div className="Toolbar sdv-toolbar-border flex-horizontal">
    <div
      className={cn("Toolbar--tool sdv-tool-border", {
        selected: props.selectedToolId === "axe"
      })}
      // tslint:disable-next-line:jsx-no-lambda
      onClick={() => {
        props.selectTool("axe");
      }}
    >
      <img src="/images/pick-axe.png" />
    </div>
    <div className="Toolbar--tool sdv-tool-border" />
    <div className="Toolbar--tool sdv-tool-border" />
    <div className="Toolbar--tool sdv-tool-border" />
    <div className="Toolbar--tool sdv-tool-border" />
    <div className="Toolbar--tool sdv-tool-border" />
    <div className="Toolbar--tool sdv-tool-border" />
  </div>
);

export default Toolbar;
