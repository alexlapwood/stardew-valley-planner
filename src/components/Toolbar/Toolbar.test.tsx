import { render } from "enzyme";
import * as React from "react";

import Toolbar from "./Toolbar";

it("renders the app without crashing", () => {
  const wrapper = render(<Toolbar images={[]} selectTool={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
