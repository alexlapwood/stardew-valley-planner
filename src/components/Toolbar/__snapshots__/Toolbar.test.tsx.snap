// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app without crashing 1`] = `
<div
  class="Toolbar sdv-toolbar-border flex-horizontal"
>
  <div
    class="Toolbar--tool sdv-tool-border"
  >
    <img
      src="/images/pick-axe.png"
    />
  </div>
  <div
    class="Toolbar--tool sdv-tool-border"
  />
  <div
    class="Toolbar--tool sdv-tool-border"
  />
  <div
    class="Toolbar--tool sdv-tool-border"
  />
  <div
    class="Toolbar--tool sdv-tool-border"
  />
  <div
    class="Toolbar--tool sdv-tool-border"
  />
  <div
    class="Toolbar--tool sdv-tool-border"
  />
</div>
`;
